paddle/cinn/optim/ir_copy.cc

```diff
@@ -419,19 +421,19 @@ struct IRCopyVisitor : public ir::IRVisitorBase<Expr> {
     }
   }
 
-#define OP_BINARY_HANDLE(op__)               \
-  Expr Visit(const ir::op__* op) override {  \
-    auto a = IRVisitorBase::Visit(&op->a()); \
-    auto b = IRVisitorBase::Visit(&op->b()); \
-    return op__::Make(a, b);                 \
+#define OP_BINARY_HANDLE(op__)                               \
+  Expr Visit(const ir::op__* op) override {                  \
+    auto a = IRVisitorRequireReImplVisitor::Visit(&op->a()); \
+    auto b = IRVisitorRequireReImplVisitor::Visit(&op->b()); \
+    return op__::Make(a, b);                                 \
   }
   NODETY_BINARY_OP_FOR_EACH(OP_BINARY_HANDLE)
 #undef OP_BINARY_HANDLE
 
-#define OP_UNARY_HANDLE(op__)                \
-  Expr Visit(const op__* op) override {      \
-    auto v = IRVisitorBase::Visit(&op->v()); \
-    return op__::Make(v);                    \
+#define OP_UNARY_HANDLE(op__)                                \
+  Expr Visit(const op__* op) override {                      \
+    auto v = IRVisitorRequireReImplVisitor::Visit(&op->v()); \
```

Comment:
Why are 2 Visitors in name?

Using `IrVisitorRequireReImpl`or `IrRequireReImplVisitor`?

Reply:
done